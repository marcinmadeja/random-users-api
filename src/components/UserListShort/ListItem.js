import React from 'react';
import { typesUserListItem } from 'constants/propTypes/userPropTypes';

import { ListItem, ListItemText } from 'material-ui/List';
import Avatar from 'material-ui/Avatar';

const propTypes = {
  user: typesUserListItem.isRequired,
};

const UserListShortItem = ({ user }) => {
  const { 
    picture: { thumbnail: avatar },
    name: { first, last },
    email,
  } = user;

  return (
    <ListItem button>
      <Avatar>
        <img src={avatar} alt="thumb" />
      </Avatar>
      <ListItemText primary={`${first} ${last}`} secondary={email} />
    </ListItem>
  );
};

UserListShortItem.propTypes = propTypes;

export default UserListShortItem;
