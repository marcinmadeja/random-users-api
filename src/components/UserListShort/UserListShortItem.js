import React from 'react';
import PropTypes from 'prop-types';

import { ListItem, ListItemText } from 'material-ui/List';
import Avatar from 'material-ui/Avatar';
import FolderIcon from 'material-ui-icons/Folder';

const UserListShortItem = ({ user }) => {
  const { 
    picture: { thumbnail: avatar },
    name: { first, last },
    email,
  } = user;

  return (
    <ListItem button>
      <Avatar>
        <img src={avatar} alt="thumb" />
      </Avatar>
      <ListItemText primary={`${first} ${last}`} secondary={email} />
    </ListItem>
  );
};

export default UserListShortItem;
