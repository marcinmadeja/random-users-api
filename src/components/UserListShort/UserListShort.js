import React, { Component } from 'react';
import { typesUserList } from 'constants/propTypes/userPropTypes';

import List from 'material-ui/List';
import UserListShortItem from './UserListShortItem';

const propTypes = {
  users: typesUserList,
};

const defaultProps = {
  users: [],
};

class UserListShorter extends Component {
  renderList() {
    const { users } = this.props;
    return (
      <List>
        {users.map(user => <UserListShortItem key={user.email} user={user} />)}
      </List>
    );
  }

  render() {
    return (
      <div>
        {this.renderList()}
      </div>
    );
  }
}

UserListShorter.propTypes = propTypes;
UserListShorter.defaultProps = defaultProps;

export default UserListShorter;
