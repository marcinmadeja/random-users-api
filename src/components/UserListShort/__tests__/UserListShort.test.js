import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import { usersShortList } from 'tests/variables';
import UserListShort from '../UserListShort';

describe('UserList', () => {
  const component = renderer.create(<UserListShort users={usersShortList} />);
  
  it('renders without crashing with no users', () => {
    shallow(<UserListShort />);
  });

  it('render correctly with four users', () => {
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
