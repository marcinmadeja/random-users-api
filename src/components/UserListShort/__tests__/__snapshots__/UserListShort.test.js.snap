// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList render correctly with four users 1`] = `
<div>
  <li
    className="MuiButtonBase-root-11 MuiListItem-root-1 MuiListItem-gutters-8 MuiListItem-button-9 MuiListItem-default-4"
    disabled={false}
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiAvatar-root-13 MuiAvatar-colorDefault-14"
    >
      <img
        alt="thumb"
        src="https://randomuser.me/api/portraits/thumb/men/43.jpg"
      />
    </div>
    <div
      className="MuiListItemText-root-16"
    >
      <h3
        className="MuiTypography-root-21 MuiTypography-subheading-28 MuiListItemText-text-19"
      >
        babür duygulu
      </h3>
      <p
        className="MuiTypography-root-21 MuiTypography-body1-30 MuiTypography-colorSecondary-42 MuiListItemText-text-19"
      >
        babür.duygulu@example.com
      </p>
    </div>
    <span
      className="MuiTouchRipple-root-45"
    />
  </li>
  <li
    className="MuiButtonBase-root-11 MuiListItem-root-1 MuiListItem-gutters-8 MuiListItem-button-9 MuiListItem-default-4"
    disabled={false}
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiAvatar-root-13 MuiAvatar-colorDefault-14"
    >
      <img
        alt="thumb"
        src="https://randomuser.me/api/portraits/thumb/men/46.jpg"
      />
    </div>
    <div
      className="MuiListItemText-root-16"
    >
      <h3
        className="MuiTypography-root-21 MuiTypography-subheading-28 MuiListItemText-text-19"
      >
        daniel polon
      </h3>
      <p
        className="MuiTypography-root-21 MuiTypography-body1-30 MuiTypography-colorSecondary-42 MuiListItemText-text-19"
      >
        daniel.polon@example.com
      </p>
    </div>
    <span
      className="MuiTouchRipple-root-45"
    />
  </li>
  <li
    className="MuiButtonBase-root-11 MuiListItem-root-1 MuiListItem-gutters-8 MuiListItem-button-9 MuiListItem-default-4"
    disabled={false}
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiAvatar-root-13 MuiAvatar-colorDefault-14"
    >
      <img
        alt="thumb"
        src="https://randomuser.me/api/portraits/thumb/men/70.jpg"
      />
    </div>
    <div
      className="MuiListItemText-root-16"
    >
      <h3
        className="MuiTypography-root-21 MuiTypography-subheading-28 MuiListItemText-text-19"
      >
        nooa joki
      </h3>
      <p
        className="MuiTypography-root-21 MuiTypography-body1-30 MuiTypography-colorSecondary-42 MuiListItemText-text-19"
      >
        nooa.joki@example.com
      </p>
    </div>
    <span
      className="MuiTouchRipple-root-45"
    />
  </li>
  <li
    className="MuiButtonBase-root-11 MuiListItem-root-1 MuiListItem-gutters-8 MuiListItem-button-9 MuiListItem-default-4"
    disabled={false}
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiAvatar-root-13 MuiAvatar-colorDefault-14"
    >
      <img
        alt="thumb"
        src="https://randomuser.me/api/portraits/thumb/men/1.jpg"
      />
    </div>
    <div
      className="MuiListItemText-root-16"
    >
      <h3
        className="MuiTypography-root-21 MuiTypography-subheading-28 MuiListItemText-text-19"
      >
        jerome morris
      </h3>
      <p
        className="MuiTypography-root-21 MuiTypography-body1-30 MuiTypography-colorSecondary-42 MuiListItemText-text-19"
      >
        jerome.morris@example.com
      </p>
    </div>
    <span
      className="MuiTouchRipple-root-45"
    />
  </li>
</div>
`;
